### PR TITLE
docs(readme): add contributing

### PR DESCRIPTION
**Pull request by BlackHole1**

### CODE REVIEW

**Review by hyrious — DISMISSED** [review]

File: README-zh.md [hunk 1]
```diff
@@ -100,6 +100,18 @@ yarn run start
 - [环境变量值参考](docs/env/README-zh.md)
 - [调试 Flat](docs/debugging/README-zh.md)
 
+## 贡献
+
+Flat 的开发过程一直在 `GitHub` 公开地进行。
```
Comment (LitoMore) [inline README-zh.md]:
```suggestion
Flat 的开发过程一直在 GitHub 公开地进行。
```

File: README-zh.md [hunk 2]
```diff
@@ -100,6 +100,18 @@ yarn run start
...
+
+我们希望通过这种方式持续开发完善 Flat，为社区提供一套开放的在线实时互动教室。
+
+我们非常欢迎来自社区的贡献，无论是通过 `issues` 参与反馈与讨论，还是通过 `pull requests` 为 Flat 提供改进优化与修复问题，你的任何帮助对我们都非常重要。
```
Comment (LitoMore) [inline README-zh.md]:
```suggestion
我们非常欢迎来自社区的贡献，无论是通过 [issues](https://github.com/netless-io/flat/issues) 参与反馈与讨论，还是通过 [pull requests](https://github.com/netless-io/flat/pulls) 为 Flat 提供改进优化与修复问题，你的任何帮助对我们都非常重要。
```

File: README-zh.md [hunk 3]
```diff
@@ -100,6 +100,18 @@ yarn run start
...
+
+你也可以通过点击下方 `Slack` 链接随时联系我们。
```
Comment (LitoMore) [inline README-zh.md]:
```suggestion
你也可以通过 [Slack](https://github.com/netless-io/flat/issues/926) 随时联系我们。
```

File: README-zh.md [hunk 4]
```diff
@@ -100,6 +100,18 @@ yarn run start
...
+
+[💬 加入 Slack 讨论组][join-flat-slack]
+
```
Comment (LitoMore) [inline README-zh.md]:
```suggestion

```

File: README-zh.md [hunk 5]
```diff
@@ -118,6 +130,8 @@ yarn run start
 
 当使用 Flat 或其他 GitHub 徽标时，请务必遵循 [GitHub 徽标指南][github-logo]。
 
+[join-flat-slack]: https://github.com/netless-io/flat/issues/926
+
```
Comment (LitoMore) [inline README-zh.md:132]:
```suggestion
```

**Comment by LitoMore** [comment]
We could drop this section from the readme, and create a `CONTRIBUTING.md` instead.

From a contributing guide, we should let coders know what is our expected contribution. And how to align their actions with our standard.

If we want to mention something, just use a link `[text](#)` to mention it. It's unnecessary to use anchors to jump another section. The `see links below` or `see here` related anchors just made our documentation hard to read.

Here is an example: https://github.com/fanfoujs/xiaofan-wechat/blob/main/CONTRIBUTING.md

### Other contributing guides, FYI

- [npm](https://github.com/npm/cli/blob/latest/CONTRIBUTING.md)
- [React](https://reactjs.org/docs/how-to-contribute.html)
- [XO](https://github.com/xojs/xo/blob/main/contributing.md)

**Comment by BlackHole1** [comment]
> We could drop this section from the readme, and create a `CONTRIBUTING.md` instead.
> 
> From a contributing guide, we should let coders know what is our expected contribution. And how to align their actions with our standard.
> 
> If we want to mention something, just use a link `[text](#)` to mention it. It's unnecessary to use anchors to jump another section. The `see links below` or `see here` related anchors just made our documentation hard to read.
> 
> Here is an example: https://github.com/fanfoujs/xiaofan-wechat/blob/main/CONTRIBUTING.md
> ### Other contributing guides, FYI
> 
> * [npm](https://github.com/npm/cli/blob/latest/CONTRIBUTING.md)
> * [React](https://reactjs.org/docs/how-to-contribute.html)
> * [XO](https://github.com/xojs/xo/blob/main/contributing.md)


Do you have any comments?

cc: @crimx @Leooeloel

**Comment by crimx** [comment]
LGTM

**Comment by Leooeloel** [comment]
> > We could drop this section from the readme, and create a `CONTRIBUTING.md` instead.
> > From a contributing guide, we should let coders know what is our expected contribution. And how to align their actions with our standard.
> > If we want to mention something, just use a link `[text](#)` to mention it. It's unnecessary to use anchors to jump another section. The `see links below` or `see here` related anchors just made our documentation hard to read.
> > Here is an example: https://github.com/fanfoujs/xiaofan-wechat/blob/main/CONTRIBUTING.md
> > ### Other contributing guides, FYI
> > 
> > * [npm](https://github.com/npm/cli/blob/latest/CONTRIBUTING.md)
> > * [React](https://reactjs.org/docs/how-to-contribute.html)
> > * [XO](https://github.com/xojs/xo/blob/main/contributing.md)
> 
> Do you have any comments?
> 
> cc: @crimx @Leooeloel

LGTM

**Comment by BlackHole1** [comment]
I have improved this PR in line with the review comments. Please re-review

cc: @LitoMore @crimx @Leooeloel @hyrious @Cheerego7 @Cyberhan123

**Review by crimx — APPROVED** [review]

**Review by Cheerego7 — APPROVED** [review]

**Review by syt-honey — APPROVED** [review]

**Review by syt-honey — APPROVED** [review]

**Review by hyrious — APPROVED** [review]

**Review by syt-honey — APPROVED** [review]

**Review by LitoMore — APPROVED** [review]
Usually, we put the `CONTRIBUTING.md` file [to the root directory](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#adding-a-contributing-file).

Since we moved this file to a sub-directory, we should let the developer find our contributing guide at first glance.

My suggestion is to use another directory or repository to store our non-English documentations. Just like [AVA](https://github.com/avajs/ava/blob/main/.github/CONTRIBUTING.md) or [Electron](https://github.com/electron/electron#documentation-translations) did (I'd prefer AVA's).

Another solution is adding `## Contribiting` to our readme like this below:

```markdown
## Contributing

See [CONTRIBUTING.md](path/to/CONTRIBUTING.md).
```

We could do this in another PR if we want this.

**Comment by BlackHole1** [comment]
> Usually, we put the `CONTRIBUTING.md` file [to the root directory](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/setting-guidelines-for-repository-contributors#adding-a-contributing-file).
> 
> Since we moved this file to a sub-directory, we should let the developer find our contributing guide at first glance.
> 
> My suggestion is to use another directory or repository to store our non-English documentations. Just like [AVA](https://github.com/avajs/ava/blob/main/.github/CONTRIBUTING.md) or [Electron](https://github.com/electron/electron#documentation-translations) did (I think AVA's solution is better).
> 
> Another solution is adding `## Contribiting` to our read like this below:
> 
> ```gfm
> ## Contributing
> 
> See [CONTRIBUTING.md](path/to/CONTRIBUTING.md).
> ```
> 
> We could do this in another PR if we want this.

I probably would have preferred to put together.
One problem with separate storage is that the synchronisation is not timely, this problem arises with ava's translation.

Personally, I think it is a better practice to place the translated file next to the source file

There are not many such exceptions, so far known: `README` and `CONTRIBUTING`

For the rest of the documentation, we can put it in `https://github.com/netless-io/flat/tree/main/docs`

cc: @netless-io/developers

**Review by crimx — APPROVED** [review]

**Comment by LitoMore** [comment]
Agree. I'm OK with storing them in the current repo or another one. Just put them together.

My main point is to let us avoid this kind of file structure below in the root directory, and put [community profiles](https://docs.github.com/en/communities/setting-up-your-project-for-healthy-contributions/about-community-profiles-for-public-repositories) to the correct directory.

```
CONTRIBUTING.md
CONTRIBUTING-zh.md
README.md
README-zh.md
SECURITY.md
SECURITY-zh.md
```

**Comment by BlackHole1** [comment]
`README` and `CONTRIBUTING` / `SECURITY` is github feature.

It may not be a good idea to put it in another location.

**Comment by LitoMore** [comment]
> README and CONTRIBUTING / SECURITY is the GitHub feature.

It's proposal, not feature. How to organize them is up to us. Our target is to make developers can find our documentation easily.

**Comment by BlackHole1** [comment]
We have discussed this in the slack and the results are as follows:

move non-English documentation to the sub-directory (e.g: `docs/`). Just leave the English version in the root directory

**Pull request merged**